modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/failures.rb

```diff
@@ -37,6 +37,20 @@ module Failures
             storage_error = ::Storages::StorageError.new(code:, log_message:, data:)
             ServiceResult.failure(result: code, errors: storage_error)
           end
+
+          ErrorData = ->(response:, source:) do
+            payload =
+              case response
+              in { content_type: { mime_type: "application/json" } }
+                response.json
+              in { content_type: { mime_type: "text/xml" } }
+                response.xml
+              else
+                response.body.to_s
+              end
+
```

Comment:
🟢 Since this just extract the payload from the response, I wonder if naming it to `ErrorDataPayload` or `PayloadExtractor` would make its intent clearer.


I'm not dying on this hill tho.

Reply:
I struggled with the naming, too.